src/components/EmojiPicker/index.js

```diff
@@ -130,70 +108,34 @@ class EmojiPicker extends React.Component {
-                    isVisible={this.state.isEmojiPickerVisible}
-                    onClose={this.hideEmojiPicker}
-                    onModalShow={this.focusEmojiSearchInput}
-                    onModalHide={this.props.onModalHide}
-                    hideModalContentWhileAnimating
-                    animationInTiming={1}
-                    animationOutTiming={1}
-                    anchorPosition={{
-                        bottom: this.props.windowHeight - this.state.emojiPopoverAnchorPosition.vertical,
-                        left: this.state.emojiPopoverAnchorPosition.horizontal - CONST.EMOJI_PICKER_SIZE,
-                    }}
-                >
-                    <EmojiPickerMenu
-                        onEmojiSelected={this.selectEmoji}
-                        ref={el => this.emojiSearchInput = el}
-                        preferredSkinTone={this.props.preferredSkinTone}
-                        updatePreferredSkinTone={this.setPreferredSkinTone}
-                        frequentlyUsedEmojis={this.props.frequentlyUsedEmojis}
-                    />
-                </Popover>
-                <Pressable
-                    style={({hovered, pressed}) => ([
-                        styles.chatItemEmojiButton,
-                        StyleUtils.getButtonBackgroundColorStyle(getButtonState(hovered, pressed)),
-                    ])}
-                    ref={el => this.emojiPopoverAnchor = el}
-                    onLayout={this.measureEmojiPopoverAnchorPosition}
-                    onPress={this.showEmojiPicker}
-                    disabled={this.props.isDisabled}
-                >
-                    {({hovered, pressed}) => (
-                        <Tooltip text={this.props.translate('reportActionCompose.emoji')}>
-                            <Icon
-                                src={Expensicons.Emoji}
-                                fill={StyleUtils.getIconFillColor(getButtonState(hovered, pressed))}
-                            />
-                        </Tooltip>
-                    )}
-                </Pressable>
-            </>
+            <PopoverWithMeasuredContent
+                isVisible={this.state.isEmojiPickerVisible}
+                onClose={this.hideEmojiPicker}
+                onModalShow={this.focusEmojiSearchInput}
+                onModalHide={this.onModalHide}
+                hideModalContentWhileAnimating
+                animationInTiming={1}
+                animationOutTiming={1}
+                anchorPosition={{
+                    vertical: this.state.emojiPopoverAnchorPosition.vertical,
+                    horizontal: this.state.emojiPopoverAnchorPosition.horizontal,
+                }}
+                anchorOrigin={{
+                    horizontal: CONST.MODAL.ANCHOR_ORIGIN_HORIZONTAL.LEFT,
+                    vertical: CONST.MODAL.ANCHOR_ORIGIN_VERTICAL.BOTTOM,
+                }}
+                measureContent={this.measureContent}
+            >
```

Comment:
`PopoverWithMeasuredContent` is creating a delay in rendering the menu. But we need it to place the anchor properly. 
 Now do this to remove the delay. 
 1. We know the dimensions of the picker beforehand so add two props width and height to `PopoverWithMeasuredContent`. 
 2. When passed, these props will skip content measuring and immediately render the menu when `isVisible` is set to `true. 
 
 Or create a new component something like `PopoverWithStaicContent` which does step 1 and does the anchor position adjustments. `PopoverWithMeasuredContent` will use this and pass down all the props as well as width and height that it has measured.

Reply:
I've added a `popoverDimensions`, can you check if that is what was expected?

Reply:
Have you tested it before I test it? Did you check if PopoverWithMeasuredContent is not measuring the dimensions when popoverDimensions is provided?

Reply:
yeah, you have to make sure that `isContentMeasured` is true when dimensions are provided in the `getDerivedStateFromProps`. 

And yes if the picker doesn't stay at the correct position, it means that content is not being measured now. this is what we want first. 
Then you have to specify the height to picker as explained in further comments and add that to styles and then it will start to place correctly.

Reply:
> yeah, you have to make sure that isContentMeasured is true when dimensions are provided in the getDerivedStateFromProps.

Should we move `this.popoverWidth` to state?

Reply:
no need. You can use the props in `getDerivedStateFromProps`. Do you see an issue there?